minerl/env/core.py

```diff
@@ -205,7 +205,12 @@ def init(self):
             xml = f.read()
         # Todo: This will fail when using a remote instance manager.
         xml = xml.replace('$(MISSIONS_DIR)', missions_dir)
-        xml = xml.replace('$(ENV_NAME)', self.spec.id)
+
+        if self.spec is not None:
+            xml = xml.replace('$(ENV_NAME)', self.spec.id)
+        else:
+            xml = xml.replace('$(ENV_NAME)', 'Custom MineRL Env')
```

Comment:
Looks fine, if you could change the string to 'MineRLCustomEnv'  and move it to a global constant (
`MINERL_CUSTOM_ENV_ID='MineRLCustomEnv'`) at the top of the file, then this is good to go!